creator-node/src/routes/nodeSync.js

```diff
@@ -159,69 +168,60 @@ module.exports = function (app) {
-      }
 
-      for (let audiusUser of fetchedCNodeUser.audiusUsers) {
-        await models.AudiusUser.create({
-          audiusUserUUID: audiusUser.audiusUserUUID,
-          cnodeUserUUID: fetchedCnodeUserUUID,
-          blockchainId: audiusUser.blockchainId,
-          metadataJSON: audiusUser.metadataJSON,
-          metadataFileUUID: audiusUser.metadataFileUUID,
-          coverArtFileUUID: audiusUser.coverArtFileUUID,
-          profilePicFileUUID: audiusUser.profilePicFileUUID
-        }, { transaction: t })
-        req.logger.info(redisKey, `created audiusUser for audiusUserUUID ${audiusUser.audiusUserUUID}`)
-      }
+      // Files with trackUUIDs cannot be created until tracks have been created,
+      // but tracks cannot be created until metadata and cover art files have been created.
+      const trackFiles = fetchedCNodeUser.files.filter(file => file.trackUUID != null)
+      const nonTrackFiles = fetchedCNodeUser.files.filter(file => file.trackUUID == null)
+      await models.File.bulkCreate(nonTrackFiles.map(file => ({
+        fileUUID: file.fileUUID,
+        trackUUID: null,
+        cnodeUserUUID: fetchedCnodeUserUUID,
+        multihash: file.multihash,
+        sourceFile: file.sourceFile,
+        storagePath: file.storagePath
+      })), { transaction: t })
+      req.logger.info(redisKey, 'created all non-track files')
+
+      await models.Track.bulkCreate(fetchedCNodeUser.tracks.map(track => ({
+        trackUUID: track.trackUUID,
+        blockchainId: track.blockchainId,
+        cnodeUserUUID: fetchedCnodeUserUUID,
+        metadataJSON: track.metadataJSON,
+        metadataFileUUID: track.metadataFileUUID,
+        coverArtFileUUID: track.coverArtFileUUID
+      })), { transaction: t })
+      req.logger.info(redisKey, 'created all tracks')
+
+      // Save all track files to disk
+      await Promise.all(trackFiles.map(trackFile => saveFileForMultihash(req, trackFile.multihash, trackFile.storagePath)))
+      req.logger.info('save all track files to disk/ipfs')
+
+      // Save all track files to db
+      await models.File.bulkCreate(trackFiles.map(trackFile => ({
+        fileUUID: trackFile.fileUUID,
+        trackUUID: trackFile.trackUUID,
+        cnodeUserUUID: fetchedCnodeUserUUID,
+        multihash: trackFile.multihash,
+        sourceFile: trackFile.sourceFile,
+        storagePath: trackFile.storagePath
+      })), { transaction: t })
+      req.logger.info('saved all track files to db')
+
+      await models.AudiusUser.bulkCreate(fetchedCNodeUser.audiusUsers.map(audiusUser => ({
+        audiusUserUUID: audiusUser.audiusUserUUID,
+        cnodeUserUUID: fetchedCnodeUserUUID,
+        blockchainId: audiusUser.blockchainId,
+        metadataJSON: audiusUser.metadataJSON,
+        metadataFileUUID: audiusUser.metadataFileUUID,
+        coverArtFileUUID: audiusUser.coverArtFileUUID,
+        profilePicFileUUID: audiusUser.profilePicFileUUID
+      })), { transaction: t })
```

Comment:
If you wanted, you could put some of the async commands above in parallel. ie 
`
await Promise.all([
  models.File.bulkCreate(nonTrackFiles...),
  models.File.bulkCreate(fetchCNodeUsers...),
])
await Promise.all(trackfiles.map(saveFiles))
await Promise.all([
  models.File.bulkCreate(trackFiles...),
  models.File.bulkCreate(AudiusUsers...),
])
`
Only a small optimization, so no need to change, looking good!